addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarImporter.java

```diff
@@ -0,0 +1,166 @@
+                Integer.valueOf(
+                        harEntry.getCustomFields()
+                                .getCustomFieldValue(HarUtils.MESSAGE_ID_CUSTOM_FIELD));
+        try {
+            httpMessage.setHistoryRef(new HistoryReference(historyId.intValue()));
+        } catch (DatabaseException | HttpMalformedHeaderException e) {
+            // Ignore.
+        }
+    }
+
+    private static void setHttpResponse(HarResponse harResponse, HttpMessage message)
+            throws HttpMalformedHeaderException {
+        StringBuilder strBuilderResHeader = new StringBuilder();
+
+        // empty responses without status code are possible
+        if (harResponse.getStatus() == 0) {
+            return;
+        }
+
+        strBuilderResHeader
+                .append(harResponse.getHttpVersion())
+                .append(' ')
+                .append(harResponse.getStatus())
+                .append(' ')
+                .append(harResponse.getStatusText())
+                .append("\r\n");
+
+        for (HarHeader harHeader : harResponse.getHeaders().getHeaders()) {
+            strBuilderResHeader
+                    .append(harHeader.getName())
+                    .append(": ")
+                    .append(harHeader.getValue())
+                    .append("\r\n");
+        }
+        strBuilderResHeader.append("\r\n");
+
+        HarContent harContent = harResponse.getContent();
+        message.setResponseHeader(new HttpResponseHeader(strBuilderResHeader.toString()));
+        if (harContent != null) {
+            message.setResponseBody(new HttpResponseBody(harContent.getText()));
+        }
+    }
+
+    static boolean importHarFile(File file) {
+        try {
+            processMessages(file);
+            return true;
+        } catch (IOException e) {
+            LOG.error(e);
+            return false;
+        }
+    }
+
+    public static void processMessages(File file) throws IOException {
+        List<HttpMessage> messages =
+                HarImporter.getHttpMessages(new HarFileReader().readHarFile(file));
+        messages.forEach(message -> persistMessage(message));
+    }
+
+    private static void persistMessage(HttpMessage message) {
+        HistoryReference historyRef;
+
+        if (message.getHistoryRef() == null) {
+            try {
+                historyRef =
+                        new HistoryReference(
+                                Model.getSingleton().getSession(),
+                                HistoryReference.TYPE_ZAP_USER,
+                                message);
+            } catch (Exception e) {
+                LOG.warn(e.getMessage(), e);
+                return;
+            }
+        } else {
+            historyRef = message.getHistoryRef();
+        }
+
+        ExtensionHistory extHistory =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class);
+        if (extHistory != null) {
+            EventQueue.invokeLater(
```

Comment:
This needs to be changed, if the view is not initialised it should not execute in the EDT.